File: vermyth/adapters/a2a/agent_card.py
from __future__ import annotations

from vermyth.adapters.a2a.types import AgentCard, AgentSkill


def build_agent_card(tool_definitions: list[dict]) -> AgentCard:
    skills: list[AgentSkill] = []
    for tool in tool_definitions:
        skills.append(
            AgentSkill(
                id=str(tool.get("name", "")),
                name=str(tool.get("name", "")),
                description=str(tool.get("description", "")),
            )
        )
    return AgentCard(
        name="vermyth",
        version="0.1.0",
        description="Semantic IR runtime with MCP tool surface.",
        skills=skills,
    )

File: typescript/src/adapters/a2a/agent-card.ts
/**
 * Agent card for `/.well-known/agent*.json` — mirrors Python `build_agent_card`.
 */
export interface AgentSkill {
  id: string;
  name: string;
  description: string;
}

export interface AgentCard {
  name: string;
  version: string;
  description: string;
  skills: AgentSkill[];
}

export function buildAgentCard(toolDefinitions: Array<Record<string, unknown>>): AgentCard {
  const skills: AgentSkill[] = [];
  for (const tool of toolDefinitions) {
    skills.push({
      id: String(tool["name"] ?? ""),
      name: String(tool["name"] ?? ""),
      description: String(tool["description"] ?? ""),
    });
  }
  return {
    name: "vermyth",
    version: "0.1.0",
    description: "Semantic IR runtime with MCP tool surface.",
    skills,
  };
}
